File: src/main/java/io/github/kydzombie/link/packet/RequestLinkConnectionsPacket.java
package io.github.kydzombie.link.packet;

import io.github.kydzombie.link.Link;
import io.github.kydzombie.link.gui.screen.LinkTerminalScreenHandler;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.network.NetworkHandler;
import net.minecraft.network.packet.Packet;
import net.modificationstation.stationapi.api.entity.player.PlayerHelper;
import net.modificationstation.stationapi.api.network.packet.IdentifiablePacket;
import net.modificationstation.stationapi.api.util.Identifier;

import java.io.DataInputStream;
import java.io.DataOutputStream;

public class RequestLinkConnectionsPacket extends Packet implements IdentifiablePacket {
    @Override
    public void read(DataInputStream stream) {

    }

    @Override
    public void write(DataOutputStream stream) {

    }

    @Override
    public void apply(NetworkHandler networkHandler) {
        System.out.println("Received Request");
        PlayerEntity player = PlayerHelper.getPlayerFromPacketHandler(networkHandler);
        if (player.container instanceof LinkTerminalScreenHandler handler) {
            System.out.println("Sending");
            handler.blockEntity.updateMachineConnections();
        } else {
            System.out.println("player.container = " + player.playerContainer);
        }
    }

    @Override
    public int size() {
        return 0;
    }

    @Override
    public Identifier getId() {
        return Link.NAMESPACE.id("request_link_connections");
    }
}

File: src/main/java/com/kydzombie/link/packet/RequestLinkConnectionsPacket.kt
package com.kydzombie.link.packet

import com.kydzombie.link.Link
import com.kydzombie.link.Link.accessing
import com.kydzombie.link.block.LinkTerminalEntity
import net.minecraft.network.PacketHandler
import net.minecraft.packet.AbstractPacket
import net.modificationstation.stationapi.api.entity.player.PlayerHelper
import net.modificationstation.stationapi.api.network.packet.IdentifiablePacket
import net.modificationstation.stationapi.api.util.Identifier
import java.io.DataInputStream
import java.io.DataOutputStream

class RequestLinkConnectionsPacket : AbstractPacket(), IdentifiablePacket {
    override fun read(dataInputStream: DataInputStream) = Unit
    override fun write(dataOutputStream: DataOutputStream) = Unit
    override fun apply(packetHandler: PacketHandler?) {
        val player = PlayerHelper.getPlayerFromPacketHandler(packetHandler)
        (accessing[player] as? LinkTerminalEntity?)?.sendUpdatePacket(player)
    }

    override fun length(): Int = 0

    override fun getId(): Identifier = Link.NAMESPACE.id("request_link_connections")
}
